aea/crypto/base.py

```diff
@@ -442,6 +442,49 @@ def update_with_gas_estimate(self, transaction: JSONLike) -> JSONLike:
         :param transaction: the transaction
         :return: the updated transaction
         """
+        raise NotImplementedError
+
+    def contract_method_call(
+        self, contract_instance: Any, method_name: str, **method_args: Any,
+    ) -> Optional[JSONLike]:
+        """Call a contract's method
+
+        :param contract_instance: the contract to use
+        :param method_name: the contract methof to call
+        :param method_args: the contract call parameters
+        """
+        raise NotImplementedError
+
+    def build_transaction(
+        self,
+        contract_instance: Any,
+        method_name: str,
+        method_args: Optional[Dict],
+        tx_args: Optional[Dict],
+    ) -> Optional[JSONLike]:
+        """Prepare a transaction
+
+        :param contract_instance: the contract to use
+        :param method_name: the contract methof to call
+        :param method_args: the contract parameters
+        :param tx_args: the transaction parameters
+        """
+        raise NotImplementedError
```

Comment:
Is this ok? This base method is being run in some tests from consensus-algorithms instead the Ethereum plugin version that overrides it.

Reply:
Well then you need to extend the base class for these tests